src/v/pandaproxy/rest/proxy.cc

```diff
@@ -22,51 +29,88 @@ namespace pandaproxy::rest {
+            // Will throw if auth fails
+            // The catch statements in reply.h will do error handling
+            auto auth_result = rq.service().authenticator().authenticate(
+              *rq.req);
+            auth_result.pass();
+            rq.user = credential_t{
+              auth_result.get_username(), auth_result.get_password()};
+        }
+
+        co_return co_await h(std::move(rq), std::move(rp));
+    };
+}
+
 server::routes_t get_proxy_routes() {
     server::routes_t routes;
     routes.api = ss::httpd::rest_json::name;
 
     routes.routes.emplace_back(
-      server::route_t{ss::httpd::rest_json::get_brokers, get_brokers});
+      server::route_t{ss::httpd::rest_json::get_brokers, wrap(get_brokers)});
 
     routes.routes.emplace_back(server::route_t{
-      ss::httpd::rest_json::get_topics_names, get_topics_names});
+      ss::httpd::rest_json::get_topics_names, wrap(get_topics_names)});
 
     routes.routes.emplace_back(server::route_t{
-      ss::httpd::rest_json::get_topics_records, get_topics_records});
+      ss::httpd::rest_json::get_topics_records, wrap(get_topics_records)});
 
     routes.routes.emplace_back(server::route_t{
-      ss::httpd::rest_json::post_topics_name, post_topics_name});
+      ss::httpd::rest_json::post_topics_name, wrap(post_topics_name)});
 
-    routes.routes.emplace_back(
-      server::route_t{ss::httpd::rest_json::create_consumer, create_consumer});
+    routes.routes.emplace_back(server::route_t{
+      ss::httpd::rest_json::create_consumer, wrap(create_consumer)});
 
-    routes.routes.emplace_back(
-      server::route_t{ss::httpd::rest_json::remove_consumer, remove_consumer});
+    routes.routes.emplace_back(server::route_t{
+      ss::httpd::rest_json::remove_consumer, wrap(remove_consumer)});
 
     routes.routes.emplace_back(server::route_t{
-      ss::httpd::rest_json::subscribe_consumer, subscribe_consumer});
+      ss::httpd::rest_json::subscribe_consumer, wrap(subscribe_consumer)});
 
-    routes.routes.emplace_back(
-      server::route_t{ss::httpd::rest_json::consumer_fetch, consumer_fetch});
+    routes.routes.emplace_back(server::route_t{
+      ss::httpd::rest_json::consumer_fetch, wrap(consumer_fetch)});
 
     routes.routes.emplace_back(server::route_t{
-      ss::httpd::rest_json::get_consumer_offsets, get_consumer_offsets});
+      ss::httpd::rest_json::get_consumer_offsets, wrap(get_consumer_offsets)});
 
     routes.routes.emplace_back(server::route_t{
-      ss::httpd::rest_json::post_consumer_offsets, post_consumer_offsets});
+      ss::httpd::rest_json::post_consumer_offsets,
+      wrap(post_consumer_offsets)});
 
     return routes;
 }
 
+config::binding<bool> make_bool_bind(configuration& cfg) {
+    config::property<bool> require_auth{
+      cfg,
+      "",
+      "",
+      {.needs_restart = config::needs_restart::no,
+       .visibility = config::visibility::user},
+      true};
+    return require_auth.bind();
+}
```

Comment:
This is a crash waiting to happen. What's your goal?

Reply:
Can you explain what might lead to the crash? It looks like the issue is that the binding is connected to a temporary object on the stack (`require_auth`) which is going to immediately go out of scope when the caller gets the binding.

Reply:
Exactly. The binding holds a pointer to the property, which goes out of scope.

Reply:
I think a binding can be creating directly with a value, but that's probably designed for tests only.

Also, as far as I can tell, this will always return true, but I imagine there are conditions under which it may need to be false.

Reply:
> This is a crash waiting to happen. What's your goal?

I just need to pass a binding<bool> that is always true to require auth for construction purposes. It is not used for anything, just for creating the authenticator.

Reply:
I'll go through a child class because the value constructor on binding is protected
```cpp
struct always_true : public config::binding<bool> {
  explicit always_true()
    : config::binding<bool>(true) {}
};
```